src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -267,66 +380,47 @@ SimulatePersonState KevinHallModel::simulate_person(Person &person, double shift
     double F = steady_F - (steady_F - F_0) * exp(-365.0 / tau);
     double L = steady_L - (steady_L - L_0) * exp(-365.0 / tau);
 
-    // Compute adjustment coefficient.
-    double adjust = -(a1 - b1) * (1.0 - exp(-365.0 / tau)) / (a1 * b2 - a2 * b1);
-
     // Compute body weight.
     double BW = F + L + G + W + ECF;
 
-    // Compute energy expenditure.
-    double EE =
-        (shift + K + gamma_F * F + gamma_L * L + delta * BW + TEF + AT + EI * x) / (1.0 + x);
-
-    // New simulated person state.
-    return SimulatePersonState{.H = H,
-                               .BW = BW,
-                               .PAL = PAL,
-                               .F = F,
-                               .L = L,
-                               .ECF = ECF,
-                               .G = G,
-                               .EE = EE,
-                               .EI = EI,
-                               .adjust = adjust};
+    // Set new state.
+    person.risk_factors.at("Glycogen"_id) = G;
+    person.risk_factors.at("ExtracellularFluid"_id) = ECF;
+    person.risk_factors.at("BodyFat"_id) = F;
+    person.risk_factors.at("LeanTissue"_id) = L;
+    person.risk_factors.at("Weight"_id) = BW;
 }
 
 double KevinHallModel::compute_G(double CI, double CI_0, double G_0) const {
     double k_G = CI_0 / (G_0 * G_0);
     return sqrt(CI / k_G);
 }
 
-double KevinHallModel::compute_W(double G) const { return 2.7 * G; }
-
-double KevinHallModel::compute_ECF(double EI, double EI_0, double CI, double CI_0,
-                                   double ECF_0) const {
-    double Na_b = 4000.0;
-    double Na_f = Na_b * EI / EI_0;
-    double Delta_Na_diet = Na_f - Na_b;
-    return ECF_0 + (Delta_Na_diet - xi_CI * (1.0 - CI / CI_0)) / xi_Na;
+double KevinHallModel::compute_ECF(double delta_Na, double CI, double CI_0, double ECF_0) const {
+    return ECF_0 + (delta_Na - xi_CI * (1.0 - CI / CI_0)) / xi_Na;
 }
 
-double KevinHallModel::compute_delta(double PAL, double BW, double H, unsigned int age,
-                                     core::Gender gender) const {
+double KevinHallModel::compute_delta(int age, core::Gender sex, double PAL, double BW,
+                                     double H) const {
     // Resting metabolic rate (Mifflin-St Jeor).
-    double RMR = 9.99 * BW + 6.25 * H * 100.0 - 4.92 * age;
-    RMR += gender == core::Gender::male ? 5.0 : -161.0;
+    double RMR = 9.99 * BW + 6.25 * H - 4.92 * age;
+    RMR += sex == core::Gender::male ? 5.0 : -161.0;
```

Comment:
I feel like these gender-specific values should all be bundled into a `struct` or something.

```cpp
struct FemaleConsts {
    static double my_value = 1.0;
    // etc.
};

struct MaleConsts {
    static double my_value = 2.0;
    // etc.
};
```

You could even define a helper macro for this sort of situation:

```cpp
#define SEX_VALUE(sex, name) \
((sex == core::Gender::male) ? MaleConsts::name : FemaleConsts::name)
```

Reply:
I'll leave this for #270, due to time constraints.